algorithms/clustering/observers.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from scitbx.array_family import flex
+from dials.util.observer import Observer, singleton
+from dials.algorithms.clustering import plots
+
+
+def uc_params_from_experiments(experiments):
+    uc_params = [flex.double() for i in range(6)]
+    for expt in experiments:
+        uc = expt.crystal.get_unit_cell()
+        for i in range(6):
+            uc_params[i].append(uc.parameters()[i])
+    return uc_params
+
+
+@singleton
+class UnitCellAnalysisObserver(Observer):
+
+    """
+    Observer to record unit cell clustering data and make plots.
+    """
+
+    def update(self, script):
+        """Update the data in the observer."""
+        self.data["dendrogram"] = script.unit_cell_dendrogram
```

Comment:
I'm not sure whether we agreed if cosym should also work on a single dataset. Running on a single dataset crashes at this point with
```AttributeError: 'cosym' object has no attribute 'unit_cell_dendrogram'```
Might be worth adding a try ..., except AttributeError here?